Generic intermediary text (#109)
ietf-wg-ohai/oblivious-http: draft-ietf-ohai-ohttp.md

to perform. For each request it receives, it makes a request of the oblivious
request resource that includes the same content. When it receives a response,
it sends a response to the client that includes the content of the response
from the oblivious request resource.

When forwarding a request, the proxy MUST follow the forwarding rules in
{{Section 7.6 of HTTP}}.  A generic HTTP intermediary implementation is suitable
for the purposes of serving an oblivious proxy resource, but additional care is
needed to ensure that client privacy is maintained.

Firstly, a generic implementation will forward unknown fields.  For oblivious
HTTP, a proxy SHOULD NOT forward unknown fields.  Though clients are not

> expected to include fields that might contain identifying information, a

Reviewer: ```suggestion
expected to include fields that might contain identifying information,
```

### Linkability Through Traffic Analysis {#ta}

This document assumes that all communication between different entities is

> protected by HTTPS.  This protects information about which resources are the

Reviewer 2: HTTPS is only required for client<>proxy communication to prevent this type of inference. Trying to clarify this might only complicate things, but this text seems to suggest that both hops need HTTPS. I don't know how I'd rewrite this any other way. 🤷

> interacted with and prevents a network observer from being able to trivially

Reviewer: It'd be nice to find an alternate wording to "interacted with"?

> protected by HTTPS.  This protects information about which resources are being
> interacted with and prevents a network observer from being able to trivially
> correlate messages on either side of a proxy.

Author: ```suggestion
protected by HTTPS.  This protects information about which resources are the
subject of request and prevents a network observer from being able to trivially
correlate messages on either side of a proxy.
```
